allennlp/common/testing/test_case.py

```diff
@@ -54,37 +54,14 @@ def _cleanup_archive_dir_without_logging(path: str):
         )
         self.mock_cleanup_archive_dir = self.patcher.start()
 
-    def tearDown(self):
+    def teardown_method(self):
         shutil.rmtree(self.TEST_DIR)
         self.patcher.stop()
 
 
-def parametrize(arg_names: Iterable[str], arg_values: Iterable[Iterable[Any]]):
-    """
-    Decorator to create parameterized tests.
-
-    # Parameters
-
-    arg_names : `Iterable[str]`, required.
-        Argument names to pass to the test function.
-    arg_values : `Iterable[Iterable[Any]]`, required.
-        Iterable of values to pass to each of the args.
-        The decorated test will be run for each inner iterable.
-    """
-
-    def decorator(func):
-        def wrapper(*args, **kwargs):
-            for arg_value in arg_values:
-                kwargs_extra = {name: value for name, value in zip(arg_names, arg_value)}
-                func(*args, **kwargs, **kwargs_extra)
-
-        return wrapper
-
-    return decorator
-
-
 _available_devices = ["cpu"] + (["cuda"] if torch.cuda.is_available() else [])
-multi_device = parametrize(("device",), [(device,) for device in _available_devices])
+
+multi_device = lambda f: pytest.mark.parametrize("device", _available_devices)(pytest.mark.gpu(f))
```

Comment:
The `pytest.mark.gpu(f)` ensures these tests are ran in the "GPU Checks" workflow